### PR TITLE
[1.x] Allow supplying `Executor` for `SystemWriteSide`

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR addresses #1275.

Now, `SystemSettings` allows customizing  an `Executor` which is used to post the system events in parallel. This provides an opportunity to improve the control over the available CPU resources on a server instance.

An example usage:

```java
var builder = BoundedContextBuilder.assumingTests();
var executor = ...;

builder.systemSettings()
    .enableParallelPosting()
    .useCustomExecutor(executor);
```

The issue asked for `ExecutorService` that is extended version of `Executor`. But in the scope of `SystemWriteSide` and its single implementation `DefaultSystemWriteSide`, we can do with just `Executor` as no methods beside `execute(Runnable)` are used.

The library version is bumped to ` 1.8.2.-SNAPSHOT.1`.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol Please take a look.

File: server/src/main/java/io/spine/system/server/DefaultSystemWriteSide.java [hunk 1]
```diff
@@ -61,12 +60,16 @@ final class DefaultSystemWriteSide implements SystemWriteSide {
...
-        if (system.config().postEventsInParallel()) {
-            commonPool().execute(() -> postEvent(event));
+        SystemConfig config = system.config();
+
+        if (config.postEventsInParallel()) {
+            config.postingExecutor().execute(() -> postEvent(event));
         } else {
             postEvent(event);
         }
+
```
Comment (armiol) [inline server/src/main/java/io/spine/system/server/DefaultSystemWriteSide.java:72]: Let's kill this line.

File: server/src/main/java/io/spine/system/server/SystemConfig.java [hunk 2]
```diff
@@ -67,7 +73,20 @@ public boolean includePersistentEvents() {
...
     @Override
     public boolean postEventsInParallel() {
-        return parallelPosting;
+        return nonNull(postingExecutor);
+    }
+
+    /**
+     * Returns an {@code Executor} to be used to post system events in parallel.
+     *
+     * <p>Before call the method, make sure parallel posting
```
Comment (armiol) [inline server/src/main/java/io/spine/system/server/SystemConfig.java:82]: Before calling this method...

File: server/src/test/java/io/spine/system/server/given/SystemContextSettingsTestEnv.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import static io.spine.grpc.StreamObservers.memoizingObserver;
+import static io.spine.option.EntityOption.Kind.ENTITY;
+import static io.spine.protobuf.AnyPacker.pack;
+
+/**
+ * Test environment for testing interop between
+ * {@link io.spine.system.server.SystemContext SystemContext} and
+ * {@link io.spine.system.server.SystemSettings SystemSettings}.
+ */
+public class SystemContextSettingsTestEnv {
```
Comment (armiol) [inline server/src/test/java/io/spine/system/server/given/SystemContextSettingsTestEnv.java:56]: This type looks `final` to me.

**Review by armiol — APPROVED** [review]
LGTM with some minor comments to address.

**Pull request merged**